source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+This class contains a list of `CartBundle\Domain\Cart\CartProcessorInterface` which are the access points for the shopware core and third party developers in the cart process. 
+```php
+interface CartProcessorInterface
+{
+    public function process(
+        CartBundle\Domain\Cart\Cart $cart,
+        CartBundle\Domain\Cart\ProcessorCart $processorCart,
+        CartContextInterface $context
+    );
+}
+
+```
+
+### Domain and Infrastructure layers
+We have taken great care in the design of the architecture, to decouple business logic from database and shopware dependencies.
+That means all shopware specify operations like database access, delivery information or price selections (with graduated prices or price groups) are separated into own gateways which can be replaced with other data sources.
+This layers are named **Domain** and **Infrastructure** and placed on the first level of the `CartBundle`.
+The domain layer should not have any dependencies to the shopware core. That's the infrastructure/bundle layer.
+Interactions with the shopware are defined over gateways. Applying this concept to the above product processor example leads to the following architecture:
+
+![image](img/img.002.png)
+
+
+## Price calculations
+The `CartBundle` contains at the moment the following calculation classes:
+* `\Shopware\Bundle\CartBundle\Domain\Price\PriceCalculator`
+    * Calculates a total price for a provided `PriceDefinition`
+    * Calculates the gross/net unit price and total price
+    * Uses the tax calculation services for including/excluding taxes of price
+
+* `\Shopware\Bundle\CartBundle\Domain\Price\PercentagePriceCalculator`
+    * Calculates a percentage price based on a provided collection of prices (`PriceCollection`)
+    * Sums all prices to a total amount and calculates percentage price value
+    * Calculates the percentage share of tax rules inside the provided prices and calculates the taxes percentage 
+    * Example:
+        * 100.00 € with 19% and 100.00€ with 7%
+        * 10% should be calculated 
+        * 200€ (price amount) * 10% => 20.00%
+        * 50% of the price is based on 19% tax calculation
+        * 50% of the price is based on  7% tax calculation
+        
+And the following tax calculation services:
+* `\Shopware\Bundle\CartBundle\Domain\Tax\TaxRuleCalculator`
+    * Tax calculation is based on a price with a simple tax rate
+    * Example
+        * 100.00 € should be calculated with 19% tax rate
+    
+* `\Shopware\Bundle\CartBundle\Domain\Tax\PercentageTaxRuleCalculator`
+    * Tax calculation is based on a percentage price value
+    * Example: 
+        * total price: 100.00 €
+        * 90.00€ should be calculated with 19% tax rate
+        * 10.00€ should be calculated with  7% tax rate
+
+## Extensibility concept
+All services in the CartBundle defined inside the service container, which means each service can be replaced or decorated. 
```

Comment:
All services in the CartBundle are defined inside the service container, which means each service can be replaced or embellished.